app/scripts/services/applicationGenerator.js

```diff
@@ -126,12 +126,19 @@ angular.module("openshiftConsole")
           to: {
             kind: "Service",
             name: serviceName
-          }
+          },
+          wildcardPolicy: 'None'
         }
       };
 
-      if (input.routing.host) {
-        route.spec.host = input.routing.host;
+      var host = input.routing.host;
+      if (host) {
+        if (host.startsWith('*.')) {
+          route.spec.wildcardPolicy = 'Subdomain';
+          route.spec.host = 'www' + host.substring(1);
```

Comment:
Just a thought, might be better to use some name other than `www``. We do allow for multiple routes (wildcard + non)to coexist in a single namespace ... and www is more than likely to be a name used there for non-wildcard hosts.